zerver/tests/test_presence.py

```diff
@@ -264,6 +264,15 @@ def test_ping_only(self):
         result = self.client_post("/json/users/me/presence", req)
         self.assertEqual(result.json()['msg'], '')
 
+    def test_bot_post(self):
+        # type: () -> None
+        bot = UserProfile.objects.get(email='default-bot@zulip.com')
+        bot.set_password('password')
+        bot.save()
+        self.login(bot.email, password='password')
```

Comment:
Hmm, I think actually `self.login` here should actually fail for a bot user.

Structurally, bot users shouldn't be able to set a password or otherwise authenticate (which you work around here by doing it manually), but we might want to lock that down more explicitly by having `login` always fail as well.  For a follow-up project, let's talk about fixing that, but it's not a requirement for this right now.